### PR TITLE
EVG-16584: Upgrade Jest and react-testing-library

### PR DESCRIPTION
**Pull request by SupaJoon**
[EVG-16584](https://jira.mongodb.org/browse/EVG-16584)

### Description 
Upgrade jest and react-testing-library
`@testing-library/react` is behind one major version because the [latest requires react 18](https://github.com/testing-library/react-testing-library/blob/c80809a956b0b9f3289c4a6fa8b5e8cc72d6ef6d/package.json#L64). I didn't upgrade `@testing-library/user-event` because doing so results in buttons inheriting a `pointer-events: none;`. There is an option on user events to prevent the pointer-event check but I was [unable to get it working as intended ](https://evergreen.mongodb.com/task_log_raw/spruce_ubuntu1604_test_patch_8b87b66dd7bbdc865fc82de6f30270faa76f8b9f_629dfd24850e61360e07f30a_22_06_06_13_12_17/0?type=T#L444). I think we will have more success with updating @testing-library/user-events once we upgrade to react 18 since its devDependencies rely on react 18.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -70,16 +70,16 @@
...
     "@leafygreen-ui/checkbox": "6.0.6",
-    "@leafygreen-ui/code": "9.0.0",
+    "@leafygreen-ui/code": "9.1.0",
     "@leafygreen-ui/confirmation-modal": "2.2.2",
     "@leafygreen-ui/expandable-card": "0.9.1",
     "@leafygreen-ui/icon": "11.4.0",
     "@leafygreen-ui/icon-button": "9.1.7",
     "@leafygreen-ui/inline-definition": "2.0.6",
-    "@leafygreen-ui/leafygreen-provider": "2.1.3",
+    "@leafygreen-ui/leafygreen-provider": "^2.2.0",
```
Comment (sophstad) [inline package.json:79]: Are these updates necessary now that we're not updating Code to v11? [9.1.0](https://github.com/mongodb/leafygreen-ui/blob/main/packages/code/CHANGELOG.md#910) has no dependency updates.

File: package.json [hunk 2]
```diff
@@ -70,16 +70,16 @@
...
     "@leafygreen-ui/menu": "11.0.1",
     "@leafygreen-ui/modal": "6.1.2",
-    "@leafygreen-ui/palette": "3.2.2",
+    "@leafygreen-ui/palette": "^3.3.2",
```
Comment (sophstad) [inline package.json:82]: Same as above.

File: package.json [hunk 3]
```diff
@@ -202,12 +202,13 @@
     "html-webpack-plugin": "4.5.0",
     "husky": "4.3.0",
     "identity-obj-proxy": "3.0.0",
-    "jest": "26.6.0",
-    "jest-circus": "^26.6.3",
-    "jest-junit": "12.0.0",
+    "jest": "^28.1.0",
```
Comment (sophstad) [inline package.json:205]: We have been using pinned dependencies for Jest and some of the other packages, is there any reason we can't continue doing that?

**Review by sophstad — CHANGES_REQUESTED** [review]

**Review by sophstad — APPROVED** [review]

File: package.json [hunk 4]
```diff
@@ -202,12 +202,13 @@
     "html-webpack-plugin": "4.5.0",
     "husky": "4.3.0",
     "identity-obj-proxy": "3.0.0",
-    "jest": "26.6.0",
-    "jest-circus": "^26.6.3",
-    "jest-junit": "12.0.0",
+    "jest": "28.1.0",
+    "jest-circus": "^28.1.0",
```
Comment (khelif96) [inline package.json:206]: Why are we installing jest-circus? 
image
https://www.npmjs.com/package/jest-circus

File: package.json [hunk 5]
```diff
@@ -147,12 +147,12 @@
     "@storybook/preset-create-react-app": "3.1.6",
     "@storybook/react": "6.1.18",
     "@svgr/webpack": "5.4.0",
-    "@testing-library/jest-dom": "5.11.6",
-    "@testing-library/react": "11.2.2",
-    "@testing-library/react-hooks": "3.5.0",
+    "@testing-library/jest-dom": "5.16.4",
+    "@testing-library/react": "12.1.5",
+    "@testing-library/react-hooks": "8.0.0",
```
Comment (khelif96) [inline package.json:152]: wow 5 major versions!

File: src/jestResolver.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+      // This workaround prevents Jest from considering uuid's module-based exports at all;
+      // it falls back to uuid's CommonJS+node "main" property.
+      //
+      // Once we're able to migrate our Jest config to ESM and a browser crypto
+      // implementation is available for the browser+ESM version of uuid to use (eg, via
+      // https://github.com/jsdom/jsdom/pull/3352 or a similar polyfill), this can go away.
+      if (pkg.name === "uuid" || pkg.name === "nanoid") {
+        delete pkg.exports;
+        delete pkg.module;
+      }
```
Comment (khelif96) [inline src/jestResolver.js:23]: https://github.com/uuidjs/uuid/pull/616#issuecomment-1111000582
Does this work instead?

This seems safer than deleting  exports from the package.

**Review by khelif96 — CHANGES_REQUESTED** [review]

**Review by khelif96 — APPROVED** [review]
Assuming tests pass LGTM

**Pull request merged**